olp-cpp-sdk-core/src/cache/InMemoryCache.cpp

```diff
@@ -105,6 +105,22 @@ void InMemoryCache::RemoveKeysWithPrefix(const std::string& key_prefix) {
   }
 }
 
+bool InMemoryCache::Contains(const std::string& key) {
+  std::lock_guard<std::mutex> lock{mutex_};
+  auto it = item_tuples_.Find(key);
+  if (it != item_tuples_.end()) {
+    auto expiry_time = std::get<1>(it.value());
+    if (expiry_time < time_provider_()) {
+      PurgeExpired(expiry_time);
```

Comment:
Why do we erase during checking if it exists? We should only delete if we add something new.